resources/assets/lib/utils/lang.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import Lang from 'lang.js';
+import { formatNumber } from 'utils/html';
+import { present } from 'utils/string';
+
+type Replacement = string | number | null | undefined;
+
+export interface Replacements {
+  [key: string]: Replacement;
+}
+
+export function trans(key: string, replacements: Replacements = {}, locale?: string) {
+  if (!transExists) {
+    locale = fallbackLocale;
+  }
+
+  return window.Lang.get(key, replacements, locale);
+}
+
+export function transArray(array: Replacement[], key = 'common.array_and') {
```

Comment:
this should only accept `string|number` (or maybe even just `string`) in the first place...